javascript/code-challenges/challenges-07.test.js

```diff
@@ -0,0 +1,438 @@
+
+For example, if the input is 'Welcome', the output will be:
+['Welcome', 'elcome', 'lcome', 'come', 'ome', 'me', 'e', ''].
+------------------------------------------------------------------------------------------------ */
+
+const howMuchPencil = (str) => {
+  let result = [];
+  // Solution code here...
+  for (let i = 0; i < str.length + 1; i++) {
+    result.push(str.slice(i));
+  }
+  console.log(result);
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function name wordsToCharList that, given a string as input, returns a new array where every element is a character of the input string.
+
+For example, wordsToCharList('gregor') returns ['g','r','e','g','o','r'].
+------------------------------------------------------------------------------------------------ */
+
+const wordsToCharList = (arr) => {
+  // Solution code here...
+  return arr.split("");
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+You are making a grocery list for ingredients needed in the gruffalo crumble recipe, below. Rather than taking the entire recipe, you only want a list of the item names.
+
+Write a function named listFoods that takes in the recipe and returns a new array of the food items without any amount or units. Just the name. For example, '1 cup flour' will return 'flour'.
+
+Use slice for this function, maybe more than once. The Array.indexOf() method may also be helpful.
+
+Do not use split for this function.
+------------------------------------------------------------------------------------------------ */
+
+const gruffaloCrumble = {
+  name: "How to make a Gruffalo Crumble",
+  ingredients: [
+    "1 medium-sized Gruffalo",
+    "8 pounds oats",
+    "2 pounds brown sugar",
+    "4 pounds flour",
+    "2 gallons pure maple syrup",
+    "16 cups chopped nuts",
+    "1 pound baking soda",
+    "1 pound baking powder",
+    "1 pound cinnamon",
+    "6 gallons melted butter",
+    "2 gallons fresh water",
+  ],
+  steps: [
+    "Pre-heat a large oven to 375",
+    "De-prickle the gruffalo",
+    "Sprinkle with cinnamon, sugar, flour, and nuts",
+    "Mix until evenly distributed",
+    "Grease a 3-foot x 3-foot casserole dish",
+    "Combine gruffalo compote with water to maintain moisture in the oven",
+    "Fold together remaining ingredients to make the crisp",
+    "Spread the crisp evenly over the gruffalo mixture",
+    "Bake for 12-15 hours",
+  ],
+};
+
+const listFoods = (recipe) => {
+  let result = [];
+  // Solution code here...
+  // result.push(recipe.ingredients[0].trim().split(" "));
+  for (let i = 0; i < recipe.ingredients.length; i++) {
+    result.push(
+      recipe.ingredients[i].slice(recipe.ingredients[i].lastIndexOf(" ") + 1)
+    );
+  }
```

Comment:
Great work on all the challenges so far! When you can, remove `x` in front of tests so I can run them :)

Regarding this challenge, I know you said that you're returning an arr with the last word in each string. I would try and break this down a bit into separate steps:

- You are iterating over recipe.ingredients.length (good! )
- Maybe first you can try to find the first space in the string and locate index of where that is
- Once you find the index of the first space, you can remove everything before first space AND the space itself (slice())
- Continue on, see if you need lastIndexOf ...